conf/machine/include/imx-base.inc

```diff
@@ -7,7 +7,8 @@ require conf/machine/include/utilities.inc
 
 # Set specific make target and binary suffix
 IMX_DEFAULT_BOOTLOADER = "u-boot-fslc"
-IMX_DEFAULT_BOOTLOADER_mx8 = "u-boot-imx"
+IMX_DEFAULT_BOOTLOADER_use-mainline-bsp = "u-boot-fslc"
+IMX_DEFAULT_BOOTLOADER_use-nxp-bsp = "u-boot-imx"
```

Comment:
The NXP BSP is not intended as all NXP components but instead the Linux fork, plus fixes, and Vivante binary driver, for example. The default used is indeed the U-Boot mainline as we have it being tested more often and we have more supported boards.

For NXP specific forks and versions you should use the fsl distros.

Reply:
The problem is that the `IMX_DEFAULT_BOOTLOADER` variable gets set to `u-boot-fslc`, even though the `fsl` distros specify `IMX_DEFAULT_BSP = "nxp"`.